### PR TITLE
Replace URI.escape with encode_www_form_component

### PR DESCRIPTION
**Pull request by talsafran**
Should resolve https://github.com/mapbox/mapbox-sdk-rb/issues/47

### CODE REVIEW

File: lib/mapbox.rb [hunk 1]
```diff
@@ -78,7 +78,7 @@ def self.uri_encode(params)
   end
 
   def self.url_encode(key)
-    URI.escape(key.to_s, Regexp.new("[^#{URI::PATTERN::UNRESERVED}]"))
```
Comment (talsafran) [inline lib/mapbox.rb:81]: I wasn't really sure if removing this was safe. Tried running tests locally and they did pass:

```console
$ bundle exec ruby -Itest test/all_tests.rb

[Coveralls] Set up the SimpleCov formatter.
[Coveralls] Using SimpleCov's default settings.
Loaded suite test/all_tests
Started
........................
Finished in 4.753479 seconds.
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
24 tests, 29 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
5.05 tests/s, 6.10 assertions/s
[Coveralls] Outside the CI environment, not sending data.
```

**Comment by talsafran** [comment]
@andrii-rymar @samfader hi 👋! Do you have any feedback on this PR or on @redcode's #46? We'd love to get rid of these noisy lil warnings. Thanks!

❤️:octocat:  your friends at GitHub

**Comment by redcode** [comment]
> @andrii-rymar @samfader hi wave! Do you have any feedback on this PR or on @redcode's #46? We'd love to get rid of these noisy lil warnings. Thanks!
> 
> heart:octocat: your friends at GitHub

I think you mean @redscode. I have nothing to do here.

**Comment by samfader** [comment]
Thanks @talsafran - will release it along with https://github.com/mapbox/mapbox-sdk-rb/pull/46

**Comment by samfader** [comment]
https://rubygems.org/gems/mapbox-sdk - released in 2.3.2

**Comment by talsafran** [comment]
Thanks @samfader ❤️

**Pull request merged**